mesh/mesh.cpp

```diff
@@ -11225,40 +11225,57 @@ GeometricFactors::GeometricFactors(const Mesh *mesh, const IntegrationRule &ir,
                "mixed meshes are not supported!");
    MFEM_ASSERT(mesh->GetNodes(), "meshes without nodes are not supported!");
 
-   const GridFunction *nodes = mesh->GetNodes();
-   const FiniteElementSpace *fespace = nodes->FESpace();
+   Init(*mesh->GetNodes(), d_mt);
+}
+
+GeometricFactors::GeometricFactors(const GridFunction &nodes,
+                                   const IntegrationRule &ir,
+                                   int flags, MemoryType d_mt)
+{
+   this->mesh = nullptr;
```

Comment:
I think the following makes more sense:
```suggestion
   this->mesh = nodes->FESpace()->GetMesh();;
```